desktop/apps/consignments2/client/actions.js

```diff
@@ -21,32 +26,107 @@ export function chooseArtistAndAdvance (value) {
   }
 }
 
+export function chooseLocation (location) {
+  return async (dispatch, getState) => {
+    const parseDetails = (place, status) => {
+      if (status !== window.google.maps.places.PlacesServiceStatus.OK) {
+        console.error('Unable to reach maps API', status)
+      } else {
+        const { address_components } = place
+        const city = find(address_components, (comp) => { return comp.types[0] === 'locality' })
+        const state = find(address_components, (comp) => { return comp.types[0] === 'administrative_area_level_1' })
+        const country = find(address_components, (comp) => { return comp.types[0] === 'country' })
+
+        const cityDisplay = city && city.long_name
+        const countryDisplay = country && country.long_name
+        const stateDisplay = state && state.long_name
+
+        dispatch(updateLocationInputValues(cityDisplay, stateDisplay, countryDisplay))
+      }
+    }
+
+    try {
+      if (window.google) {
+        const placesService = new window.google.maps.places.PlacesService(document.createElement('div'))
+        await placesService.getDetails({ placeId: location.place_id }, parseDetails)
+      }
+    } catch (err) {
+      console.error('error!', err)
+    }
+  }
+}
+
 export function clearArtistSuggestions () {
   return {
     type: CLEAR_ARTIST_SUGGESTIONS
   }
 }
 
+export function clearLocationSuggestions () {
+  return {
+    type: CLEAR_LOCATION_SUGGESTIONS
+  }
+}
+
 export function createSubmission () {
-  return (dispatch) => {
-    // TODO: async request to convection to create submission
-    dispatch(updateSubmission()) // update state
-    dispatch(incrementStep()) // move to next step
+  return async (dispatch, getState) => {
+    try {
+      const {
+        body: {
+          token
+        }
+      } = await request
```

Comment:
We have to get a JWT from gravity first, which is encoded using convection's `internal_secret`. We then use this to auth with convection and create a submission for the right `user_id`.